types/FrameTypes.ts

```diff
@@ -28,12 +28,23 @@ export type FrameType = {
 };
 
 export type Frame = TextFrame | ImageFrame;
+export type ImageVariableSource = {
+    assetId?: string;
+    connectorId?: string;
+    sourceType: string;
+    variableId: string;
+}
+export type ImageUrlSource = {
+    sourceType: string;
```

Comment:
create an enum ImageSourceType with `url` and `urlVariable` and then use them in the related types as `sourceType`.